database/src/main/postgres/runs/V1.5.12__write_additional_data.sql

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION runs.write_additional_data(
```

Comment:
This function does way too much stuff. It should be split into several functions.

Reply:
it's actually just 2 insert queries and 1 update, but I guess that I can break it down a little. The insert into history table can happen outside of this function, I'm gonna do that